app/models/chat_message_email_status.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+class ChatMessageEmailStatus < ActiveRecord::Base
+  self.inheritance_column = nil # Disable STI due to `type` column
+
+  STATUSES = {
+    unprocessed: 0,
+    processed: 1,
+  }
+
+  TYPES = {
+    regular: 0,
+    group_mention: 1,
+    global_mention: 2,
+    direct_mention: 3,
+  }
+
+  enum status: STATUSES
+  enum type: TYPES
```

Comment:
I'm on the fence about adding the `type` column at this time. If we're not using it yet, maybe it's best to leave it out for the current version?